File: app/src/main/java/com/zack/enderplan/injector/module/TypeDetailPresenterModule.java
package com.zack.enderplan.injector.module;

import com.zack.enderplan.view.contract.TypeDetailViewContract;

import dagger.Module;
import dagger.Provides;

@Module
public class TypeDetailPresenterModule {

    private final TypeDetailViewContract mTypeDetailViewContract;
    private final int mTypeListPosition;

    public TypeDetailPresenterModule(TypeDetailViewContract typeDetailViewContract, int typeListPosition) {
        mTypeDetailViewContract = typeDetailViewContract;
        mTypeListPosition = typeListPosition;
    }

    @Provides
    TypeDetailViewContract provideTypeDetailViewContract() {
        return mTypeDetailViewContract;
    }

    @Provides
    int provideTypeListPosition() {
        return mTypeListPosition;
    }
}

File: app/src/main/java/net/zackzhang/app/end/injector/module/TypeDetailPresenterModule.kt
package net.zackzhang.app.end.injector.module

import dagger.Module
import dagger.Provides
import net.zackzhang.app.end.view.contract.TypeDetailViewContract

@Module
class TypeDetailPresenterModule(private val mTypeDetailViewContract: TypeDetailViewContract, private val mTypeListPosition: Int) {

    @Provides
    fun provideTypeDetailViewContract() = mTypeDetailViewContract

    @Provides
    fun provideTypeListPosition() = mTypeListPosition
}
